pilopt/src/lib.rs

```diff
@@ -382,6 +383,7 @@ fn extract_constant_lookups<T: FieldElement>(pil_file: &mut Analyzed<T>) {
         .identities
         .iter_mut()
         .filter(|id| id.kind == IdentityKind::Plookup)
+        .filter(|id| id.right.selector.is_some())
```

Comment:
This is case 1.

Reply:
Wait, if there is no RHS selector, it should still work, shouldn't it?

Reply:
I think instead of this you should check that RHS selector is constant (either None or Some(constant)).

Reply:
You should change `pol_id` to something like
```
let pol_id = l_sel * (l.clone() - AlgebraicExpression::from(*r));
```
Assuming `r_sel` is 1 at least on a single row. If r_sel is None, this is true. If it's `Some(r_sel_i)`, then it depends on that value.